[mlir][vector] Refactor vector-transfer-unroll.mlir (NFC)

pabloantoniom opened the pull request:
Following @banach-space initiative of refactoring vector tests, this patch refactors `vector-transfer-unroll.mlir`:

- Unified identation of `FileCheck` commands
- Separated tests individually (previously there was a big test containing a couple of tests)
- All memref input arguments are re-named to %mem.
- All vector input arguments are re-named to %vec.
- LIT variables are update to be consistent with input arguments.
- Renamed all output arguments as %res.

banach-space commented on mlir/test/Dialect/Vector/vector-transfer-unroll.mlir:
```diff
@@ -2,363 +2,373 @@
-
-func.func @vector_gather_unroll(%arg0 : memref<?x?x?xf32>,
+// ORDER-SAME:    %[[MEM:.*]]: memref<?x?x?xf32>
+// ORDER-SAME:    %[[ARG1:.*]]: vector<6x4xindex>
+// ORDER-SAME:    %[[ARG2:.*]]: vector<6x4xi1>
+// ORDER-SAME:    %[[ARG3:.*]]: vector<6x4xf32>
```
How about more descriptive var names, e.g. `INDICES`, `MASK`, `PT`?

pabloantoniom replied:
Good catch! Fixed 👍

banach-space commented on mlir/test/Dialect/Vector/vector-transfer-unroll.mlir:
```diff
@@ -2,363 +2,373 @@
 // RUN: mlir-opt %s -test-vector-transfer-unrolling-patterns=reverse-unroll-order --split-input-file | FileCheck %s --check-prefix=ORDER
 
 // CHECK-LABEL: func @transfer_read_unroll
```
CHECK-LABEL and CHECK-SAME seem to be repeated in this file quite a lot. Consider using a "shared" label like here:
* https://github.com/llvm/llvm-project/blob/main/mlir/test/Dialect/Vector/linearize.mlir

banach-space commented on mlir/test/Dialect/Vector/vector-transfer-unroll.mlir:
Is this testing a specific pattern? If yes, could you add a comment similar to e.g.:
* https://github.com/llvm/llvm-project/blob/a50b9633357007ff886f3fd228ca4b8a9b9b9852/mlir/test/Dialect/Vector/vector-transfer-collapse-inner-most-dims.mlir#L3-L6

If this is exercising multiple patterns, I'd probably skip.

pabloantoniom replied:
It is using multiple patterns (at least `UnrollTransferReadPattern`, `UnrollTransferWritePattern` and `UnrollGatherPattern`) and the tests are mixed (not grouped like in the example you show) so I think it's better to skip it for this case.

banach-space reviewed (commented):
Thank you for helping out with this 😅 

Already looks great, but I've made a few small suggestions that could make this even greater 🙏🏻 WDYT?

pabloantoniom committed:
[mlir][vector] Refactor vector-transfer-unroll.mlir (NFC)

Overview of changes:

- All memref input arguments are re-named to %mem.
- All vector input arguments are re-named to %vec.
- LIT variables are update to be consistent with input arguments.
- Renamed all output arguments as %res.
- Unified identation of `FileCheck` commands
- Separated tests (previously there was a big test containing a couple of tests)

pabloantoniom committed:
Address banach-space comments

pabloantoniom commented:
> Thank you for helping out with this 😅
> 
> Already looks great, but I've made a few small suggestions that could make this even greater 🙏🏻 WDYT?

Looks good! I've addressed all your comments and also fixed a small detail I didn't notice before. In `transfer_read_unroll_tensor` I wrongly replaced the name of the argument with `%mem` (but it shouldn't be, because it's a tensor).

banach-space reviewed (approved):
LGTM, thank you for addressing my comments  and for improving this test 🙏🏻 

Do you have commit access? If not, can land this for you.

pabloantoniom commented:
> LGTM, thank you for addressing my comments and for improving this test 🙏🏻
> 
> Do you have commit access? If not, can land this for you.

Yes I have. Thank you for the review! 👍
